x/leverage/keeper/keeper.go

```diff
@@ -74,18 +74,27 @@ func (k Keeper) LendAsset(ctx sdk.Context, lenderAddr sdk.AccAddress, loan sdk.C
 		return err
 	}
 
-	// mint uToken
+	// determine uToken amount to mint
 	uToken, err := k.ExchangeToken(ctx, loan)
 	if err != nil {
 		return err
 	}
 
+	// mint uToken
 	uTokens := sdk.NewCoins(uToken)
 	if err = k.bankKeeper.MintCoins(ctx, types.ModuleName, uTokens); err != nil {
 		return err
 	}
 
-	if err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, lenderAddr, uTokens); err != nil {
+	if k.GetCollateralSetting(ctx, lenderAddr, uToken.Denom) {
```

Comment:
So we send `uTokens` to the module account if the `uToken.Denom` is NOT enabled as collateral?

Reply:
The opposite - collateral-enabled `uTokens` stay in the module (contributing to the lender's collateral amount), and collateral-disabled `uTokens` go in the lender's account.

Reply:
Is there a way we can refactor this so it's a bit more clear? In the case where it's disabled, we send to the module account and then back again costing the user more gas. Perhaps we move the sending to an if/else at the bottom or something.

Reply:
I see the misunderstanding - the `SendCoinsFromAccountToModule(ctx, lenderAddr, types.ModuleName, loanTokens)` is base assets and the `SendCoinsFromModuleToAccount(ctx, types.ModuleName, lenderAddr, uTokens)` is uTokens.

So the transfers are:
- In all cases, `loanTokens` go from lender to module
- When collateral disabled, `uTokens` go from module to lender
- When collateral enabled, there is no second transfer